### PR TITLE
RestoreDashboards: Correct tracking data

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**
Payload data for tracking `grafana_manage_dashboards_delete_clicked` didn't show the expected values for restore_enabled. We expect a boolean here and got null if the feature toggle wasn't enabled.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const DeleteModal = ({ onConfirm, onDismiss, selectedItems, ...props }: P
         folder: Object.keys(selectedItems.folder).length,
       },
       source: 'browse_dashboards',
-      restore_enabled: config.featureToggles.dashboardRestore,
+      restore_enabled: config.featureToggles.dashboardRestore ?? false,
```
Comment (Clarity-89) [inline public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx:30]:
```suggestion
      restore_enabled: Boolean(config.featureToggles.dashboardRestore),
```
Could we rap it into `Boolean` instead?

Comment (eledobleefe) [reply public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx:30]: @Clarity-89, is using `Boolean` a better practice? 🤔

**Review by eledobleefe — COMMENTED** [review]
I found that this property is also in [BrowseActions](https://github.com/grafana/grafana/blob/main/public/app/features/browse-dashboards/components/BrowseActions/BrowseActions.tsx#L118). You might want to modify that line also 🤷

File: public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx (hunk 1 above)
Comment (Clarity-89) [reply public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx:30]: It just looks cleaner imo.

**Review by eledobleefe — APPROVED** [review]
LGTM 👌

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**